pkg/cue/process/handle.go

```diff
@@ -235,21 +253,29 @@ func (ctx *templateContext) BaseContextFile() string {
 	}
 
 	if ctx.data != nil {
-		d, _ := json.Marshal(ctx.data)
+		d, err := json.Marshal(ctx.data)
+		if err != nil {
+			return "", err
+		}
 		buff += fmt.Sprintf("\n %s", structMarshal(string(d)))
 	}
 
-	return fmt.Sprintf("context: %s", structMarshal(buff))
+	return fmt.Sprintf("context: %s", structMarshal(buff)), nil
 }
 
 // ExtendedContextFile return cue format string of templateContext and extended secret context
 func (ctx *templateContext) ExtendedContextFile() string {
-	context := ctx.BaseContextFile()
-
+	context, err := ctx.BaseContextFile()
+	if err != nil {
+		return fmt.Sprintf("failed to convert data to application with marshal err %v", err)
```

Comment:
That is to say, it's better if we return `"", err` here.  Thin function should can be `func (ctx *templateContext) ExtendedContextFile() (string, error)`

Reply:
Thanks much! @FogDong for your kind suggestion!